File: performance/DirectoryGetFiles.java
//
// Created by Eugeny Grishul.
//
package test;

import java.nio.file.*;

public class DirectoryGetFiles {

	public static int run() throws Throwable {
		int dataDependency = 0;
		String os = System.getProperty( "os.name" );

		Path path = Paths.get( os.contains( "Windows" ) ? "c:\\windows\\fonts" : "/usr/include" );
		for( int i = 0, max = TestDispatcher.IterationCount; i < max; ++i )
			try( DirectoryStream<Path> stream = Files.newDirectoryStream( path ) ) {
				for( Path entry : stream ) {
					if( Files.isDirectory( entry ) )
						continue;
					
					String fullPath = entry.toString();
					dataDependency += fullPath.length();
				}
			}

		return dataDependency;
	}
}

File: performance/DirectoryGetFiles.cs
//
// Created by Eugeny Grishul.
//

using System;

namespace Test {
	class DirectoryGetFiles {
		public static int Run() {
			int dataDependency = 0;

			for( int i = 0, max = TestDispatcher.IterationCount; i < max; ++i ) {
#if MONO
				foreach( var entry in System.IO.Directory.GetFiles( @"/usr/include" ) ) {
#else
				foreach( var entry in System.IO.Directory.EnumerateFiles( @"c:\Windows\fonts" ) ) {
#endif
					dataDependency += entry.Length;
				}
			}

			return dataDependency;
		}
	}
}